### PR TITLE
ESMFold backend with attention visualization via OpenFold pipeline (issue #43)

### PR DESCRIPTION
**Pull request by JeevanandanRamasamy**
## Summary
Integrates a complete ESMFold backend into VizFold, enabling structure prediction and attention visualization using the `fair-esm` library loaded through OpenFold's model architecture. Outputs are written in the VizFold archive format, making them directly compatible with all existing visualization tools.

This addresses Issue #43: ESMFold inference is wired up, attention maps are extracted, and outputs are validated against the existing visualization pipeline.

### OpenFold-based ESMFold integration
- Uses `fair-esm` (`esm.pretrained.esmfold_v1()`) for checkpoint loading
- IPA key names in the checkpoint are incompatible with VizFold's OpenFold build and patched on disk before load (skipped on subsequent runs)
- GPU-enabled inference validated on PACE ICE via SLURM

### Attention extraction (fully working)
- `seq_attention` (sequence self-attention, 32 heads): captured via PyTorch forward hook on fair-esm's `Attention` class and reformatted into `ATTENTION_METADATA` before saving
- `tri_att_start` (triangle attention on pair representation, 4 heads): captured natively via VizFold's `_attention()` / `ATTENTION_METADATA` pipeline by injecting `_attention_name` and `attention_config` on the inner OpenFold `Attention` primitive
- 96 attention files per run: 48 `msa_row_attn_layer*.txt` + 48 `triangle_start_attn_layer*_residue_idx_avg.txt`

### Structure export
- Writes `structure/predicted.pdb` compatible with PyMOL and existing VizFold tools

### Metadata integration
- Generates `meta.json` consistent with VizFold archive schema (sequence hash, device, layer count, trace mode, top-k, seed)

### CLI support
- `run_pretrained_esmf.py` supports `--fasta`, `--out`, `--device`, `--trace_mode`, `--top_k`, `--layers`, `--triangle_residue_idx`
- Both `--trace_mode none` and `--trace_mode attention` validated end-to-end on ICE

### CODE REVIEW

**Review by jayvenn21 — COMMENTED** [review]
ignore (comment for reviewer access)

**Comment by jayvenn21** [comment]
notes for Jeeva's PR: i've looked through this and the attention capture is really solid. hooking block.seq_attention directly and getting the actual weights is the right approach from what i can see. also,  athe text-file format matching OpenFold output is probably the most useful piece since existing viz tools just work and as we discussed prior i think that will be the piece to add in the final deliverable.

I cleaned up the backend on feature/esmfold-backend and attention tensors are [B, H, N, N] now. if you want to port the text-file writer over you can PR against that branch

File: vizfold/backends/esmfold/inference.py [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+            ESMFOLD_CHECKPOINT_URL,
+            checkpoint_path,
+            progress=True,
+        )
+
+    logger.info("Checking ESMFold checkpoint IPA key format...")
+    ckpt = torch.load(checkpoint_path, map_location="cpu", weights_only=False)
+    state_dict = ckpt.get("model", ckpt)
+
+    needs_patch = any(
```
Comment (jayvenn21) [inline vizfold/backends/esmfold/inference.py:80]: looked here and i notice that loading the full checkpoint into cpu mem for the purpose to check key names is pretty computationally expensive. i suggest checking only the keys without loading the full tensor data. also, you could cache a flag file after the first patch

File: vizfold/backends/esmfold/inference.py [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+                log("Saving all layers")
+            self._enable_attention_capture(model, attn_map_dir, top_k,
+                                            triangle_residue_idx,
+                                            num_recycles_save)
+
+        log(f"Running ESMFold inference on {self.device}...")
+        t_start = time.perf_counter()
+
+        with torch.no_grad():
+            pdb_output = model.infer_pdb(seq)
```
Comment (jayvenn21) [inline vizfold/backends/esmfold/inference.py:280]: model.infer_pdb(seq) is a clean approach but i suggest switching to a full forward pass so that we can save predicted.pt as well. the issue with model.infer_pdb(seq)  is that we can't get the raw positions or the plddt scores
